packages/manager/apps/dedicated/client/app/ip/ip/agoraOrder/ip-ip-agoraOrder.controller.js

```diff
@@ -148,65 +150,104 @@ angular.module('Module.ip.controllers').controller(
       return 'USA';
     }
 
+    loadPrivateCloudIpOffers(serviceName) {
+      return this.$http
+        .get(`/order/cartServiceOption/privateCloud/${serviceName}`)
+        .then((data) => get(data, 'data'))
+        .then((offers) =>
+          filter(offers, {
+            family: 'ip',
+          }),
+        )
+        .then((offers) => {
+          this.ipOffers = map(offers, (offer) => {
+            const price = head(offer.prices);
+            const maximumQuantity = get(price, 'maximumQuantity') || 1;
+            return {
+              ...offer,
+              productShortName: offer.productName,
+              price: price.price,
+              duration: 'P1M', // @todo use price.duration when api is fixed
+              pricingMode: price.pricingMode,
+              maximumQuantity,
+              quantities: range(1, maximumQuantity + 1),
+              isIpBlockOffer: maximumQuantity === 1,
+            };
+          });
+        });
+    }
+
     loadIpOffers() {
       this.loading.ipOffers = true;
 
       this.model.params = {};
+      let ipOffersPromise;
 
-      const ipOffersPromise = this.IpAgoraOrder.getIpOffers(
-        this.user.ovhSubsidiary,
-      ).then((ipOffers) => {
-        let ipOfferDetails = ipOffers.map(this.createOfferDto.bind(this));
-        if (this.model.selectedService.type === PRODUCT_TYPES.vps.typeName) {
-          ipOfferDetails = ipOfferDetails
-            .filter(({ productShortName }) =>
-              productShortName.includes('failover'),
-            )
-            .map((offer) => ({
-              ...offer,
-              quantities: range(1, VPS_MAX_QUANTITY + 1),
-            }));
-        }
+      this.isPrivateCloudOffer =
```

Comment:
May be we can move this to `$onInit()` since it is being used in a lot of places... Just a suggestion :)

Reply:
i can't :/ i need the selectedService of this.model to be set, i need to wait for the user to select a model